src/bramble/client/ProjectStats.js

```diff
@@ -0,0 +1,141 @@
+define([
+    // Change this to filer vs. filer.min if you need to debug Filer
+    "thirdparty/filer/dist/filer.min",
+],function(Filer){
+    "use strict";
+
+    function ProjectStats() {};
+    
+    var _Path = Filer.Path;
+    var _root;
+
+    // keep track of project state
+    // host cache in the object should any other project states have to be added
+    var _cache = {};
+
+    var _fs = new Filer.FileSystem();
+    var _shell = new _fs.Shell();
+    ProjectStats.getFileSystem = function(){
+        return _fs;
+    };
+    // NOTE: THIS WILL DESTROY DATA! For error cases only, or to wipe the disk.
+    ProjectStats.formatFileSystem = function(flags, callback){
+        _fs = new Filer.FileSystem(flags, callback);
+
+        return _fs;
+    };
+
+    // walk the whole file tree and record each path and file size
+    ProjectStats.init = function(root, callback){
+        function addSize(path, next){
+            // If the current path is a directory, record 0 for size
+            if(path.endsWith("/")) {
+                return next();
+            }
+
+            _fs.stat(path,function(err, stats){
+                if(err){
+                    return next(err);
+                }
+                _cache[path] = stats.size;
+                next();                
+            });
+        
+        }
+
+        // reinitialize project stats
+        _root = root;
+        _cache = {};
+
+        // walk the root
+        _shell.find(_root, { exec:addSize }, function(err){
```

Comment:
Because you aren't doing anything specific with the final callback, except checking `err`, it's common to not do anything, and just use the callback passed to you instead:

```js
_shell.find(_root, { exec:addSize }, callback);
```

If you were doing something (e.g., using an object returned in the callback, changing a variable, etc.) they you'd do what you did here.